docs/managing-e2e-tests.md

```diff
@@ -0,0 +1,166 @@
+2) Updating testgrid to reflect the changes.
+3) Finalizing the changes.
+
+#### Updating test jobs
+
+This document will cover information on how to perform updates on these 3 files:
+- [kubeadm.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm.yaml "kubeadm.yaml")
+
+Holds test jobs where the kubeadm version matches the Kubernetes control-plane and the kubelet versions.
+
+- [kubeadm-upgrade.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-upgrade.yaml "kubeadm-upgrade.yaml")
+
+Holds test jobs that perform a upgrade from version X to version Y (usually `Y = X + 1`).
+
+- [kubeadm-x-on-y.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-x-on-y.yaml "kubeadm-x-on-y.yaml")
+
+Holds test jobs that run kubeadm version Y, on a control plane and kubelet version X,
+as kubeadm does support `Y = X + 1`
+
+The files can be found in the [sig-cluster-lifecycle](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/sig-cluster-lifecycle) folder.
+
+The idea is to remove old test jobs and add new ones. If Kubernetes 1.15 is about to be released soon, this means that `15 - 3 = 12` MINOR versions would go outside of the support skew of the project (the project would support `15 - 2 = 13` So, `1.12` jobs have to be removed and `1.15` jobs have to be added.
+
+- If there are `1.12 -> 1.13` upgrade jobs or `1.12 on 1.13` jobs leave them, because they still contain a supported version `1.13`.
+- Remove plain `1.12` jobs.
+- Add upgrade jobs for `1.14 -> 1.15` add `1.14 on 1.15` jobs.
+- Add plain `1.15` jobs.
+
+Each job contains some versioned labels and branches passed as properties, that are not going to be outlined in this document, but they need updates too:
+- `release-1.14`
+- `release/latest-1.14`
+- `ci/latest-bazel-1.14`
+- `latest-bazel-1.14`
+- ...
+
+#### Updating testgrid to reflect the changes
+
+Once the changes to the jobs have been made it is time to open the testgrid [configuration file](https://github.com/kubernetes/test-infra/blob/master/testgrid/config.yaml) and modify it as well .
+
+The same procedure applies here - old jobs are being removed new jobs are being added.
+
+Search and replace is your friend. Remember the job names that you just changed in the the `kubeadm*.yaml` files. Don't forget that jobs can exists under multiple dashboards in testgrid, so make sure you find all occurrences and modify them.
+
+Following the `1.15` example from earlier, a job called `ci-kubernetes-e2e-kubeadm-gce-1-12` would have to be removed from all occurrences and a new job `ci-kubernetes-e2e-kubeadm-gce-1-15` has to be added in it's place.
+
+Removed:
+```
+- name: kubeadm-gce-1.12
+  test_group_name: ci-kubernetes-e2e-kubeadm-gce-1-12
+```
+
+Added:
+```
+- name: kubeadm-gce-1.15
+  test_group_name: ci-kubernetes-e2e-kubeadm-gce-1-15
+```
+
```

Comment:
i did it in one big PR that was quite messy because it shifted things around too.
hh's PR on the other hand was missing the removel of old jobs.

https://github.com/kubernetes/test-infra/pull/9219
https://github.com/kubernetes/test-infra/pull/8142

i will add these but they are not good examples.